sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -688,11 +691,42 @@ private ConnectionOptions getConnectionOptions() {
             ? CbsAuthorizationType.SHARED_ACCESS_SIGNATURE
             : CbsAuthorizationType.JSON_WEB_TOKEN;
 
-        final ClientOptions options = clientOptions != null ? clientOptions : new ClientOptions();
         final SslDomain.VerifyMode verificationMode = verifyMode != null
             ? verifyMode
             : SslDomain.VerifyMode.VERIFY_PEER_NAME;
 
+        final ClientOptions options = new ClientOptions();
+        if (clientOptions != null) {
+            options.setApplicationId(clientOptions.getApplicationId())
+                .setHeaders(clientOptions.getHeaders());
+        }
+
+        final Map<String, String> properties = CoreUtils.getProperties(EVENTHUBS_PROPERTIES_FILE);
+        final String product = properties.getOrDefault(NAME_KEY, UNKNOWN);
+        final String clientVersion = properties.getOrDefault(VERSION_KEY, UNKNOWN);
+
+        final List<Header> headers = new ArrayList<>();
+        boolean foundName = false;
+        boolean foundVersion = false;
+        for (Header header : options.getHeaders()) {
+            if (NAME_KEY.equals(header.getName())) {
+                foundName = true;
+            } else if (VERSION_KEY.equals(header.getName())) {
+                foundVersion = true;
+            }
+
+            headers.add(header);
+        }
+
+        if (!foundName) {
+            headers.add(new Header(NAME_KEY, product));
```

Comment:
I don't think we should be using `ClientOptions` to set producet and version information. `ClientOptions` is provided by the user and we should try not to add headers that they don't specify.